### PR TITLE
faraday: init at 0.2.3

### PR DESCRIPTION
**Pull request by ProofOfKeags**
###### Motivation for this change

This adds the package `faraday` to extend the capabilities of `lnd` to report on Lightning Network channel performance.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/blockchains/faraday.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  src = fetchFromGitHub {
+    owner = "lightninglabs";
+    repo = "faraday";
+    rev = "v${version}";
+    sha256 = "16cqaslsbwda23b5n0sfppybd3ma4ch545100ydxrwac4zhrq4kq";
+  };
+
+  vendorSha256 = "1hh99nfprlmhkc36arg3w1kxby59i2l7n258cp40niv7bjn37hrq";
+
+  subPackages = ["cmd/frcli" "cmd/faraday"];
```
Comment (SuperSandro2000) [inline pkgs/applications/blockchains/faraday.nix:19]:
```suggestion
  subPackages = [ "cmd/frcli" "cmd/faraday" ];
```

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -27532,6 +27532,8 @@ in
 
   exodus = callPackage ../applications/blockchains/exodus { };
 
+  faraday = callPackage ../applications/blockchains/faraday.nix { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:27535]:
```suggestion
  faraday = callPackage ../applications/blockchains/faraday { };
```

Comment (ProofOfKeags) [reply pkgs/top-level/all-packages.nix:27535]: this is not how the rest of the file tree appears to work.

Comment (SuperSandro2000) [reply pkgs/top-level/all-packages.nix:27535]: It shouldn't be like this.

Comment (ProofOfKeags) [reply pkgs/top-level/all-packages.nix]: fixed by 6a47b6b

**Comment by SuperSandro2000** [comment]
Please add yourself as a maintainer by adding an alphabetically sorted entry to `maintainers/maintainer-list.nix` and commit this change into a separate commit with the message ``maintainers: add NAME``.

**Comment by ProofOfKeags** [comment]
in a separate commit or a separate PR altogether?

**Comment by SuperSandro2000** [comment]
> in a separate commit or a separate PR altogether?

Separate commit please.

Can you also squash all the package related commits? After that we can merge this.

**Pull request merged**